Cargo.toml

```diff
@@ -23,6 +23,10 @@ semver = "0.5.0"
 separator = "0.3.1"
 time = "0.1.35"
 
+[dependencies.clippy]
+optional = true
+version = "0.0.118"
```

Comment:
My guess would be that independent of bug fixes (which isn't easy to predict based on the utter contempt for semver in clippy's versioning), only new lints would be added rather than the scope of existing lints being expanded; this is only my intuition though, as I doubt that clippy gives any sort of guarantees about that.

You make a good point about the potential for accidentally mixing lint fixes with other PRs. While I think there might be ways to organize around this, if you'd prefer not to do this then I'm all right with forgoing it. That being said, it might be work figuring out some sort of automated way to of incorporating `cargo outdated` into our workflow (either just for clippy or all of our root dependencies). Of course, to correctly do that, we'd need to figure out how often and when we want to update our dependencies. For some dependencies (e.g. `bitflags`), it's probably not very high priority to update every time a new version comes out, but for things like `bson`, we'd ideally would want to put in the effort to try to keep the version update (like we've been doing lately by releasing a new version whenever `bson` does).